src/aatoolbox/datasources/usgs/usgs_ndvi.py

```diff
@@ -0,0 +1,924 @@
+            GeoDataFrame with row per area for stats computation.
+            If ``pd.DataFrame`` is passed, geometry column must
+            have the name ``geometry``. Passed to
+            ``aat.compute_raster_stats()``.
+        feature_col : str
+            Column in ``gdf`` to use as row/feature identifier.
+            and dates. Passed to ``aat.compute_raster_stats()``.
+        clobber : bool, default = False
+            If True, overwrites existing processed dates. If
+            the new file matches the old file, dates will be
+            reprocessed and appended to the data frame. If
+            files do not much, the old file will be replaced.
+            If False, stats are only calculated for year-dekads
+            that have not already been calculated within the
+            file. However, if False and files do not match,
+            value error will be raised.
+        **kwargs
+            Additional keyword arguments passed to
+            ``aat.computer_raster_stats()``.
+
+        Returns
+        -------
+        Path
+            The processed path
+
+        Examples
+        --------
+        >>> from aatoolbox import create_country_config, \
+        ...  CodAB, UsgsNdviSmoothed
+        >>>
+        >>> # Retrieve admin 2 boundaries for Nepal
+        >>> country_config = create_country_config(iso3="npl")
+        >>> codab = CodAB(country_config=country_config)
+        >>> npl_admin2 = codab.load(admin_level=2)
+        >>> npl_admin1 = codab.load(admin_level=2)
+        >>>
+        >>> # setup NDVI
+        >>> UsgsNdviSmoothed(
+        ...     country_config=country_config,
+        ...     coverage_area="west_africa",
+        ...     start_date=[2020, 1],
+        ...     end_date=[2020, 3]
+        ... )
+        >>> UsgsNdviSmoothed.download()
+        >>> UsgsNdviSmoothed.process()
+        >>>
+        >>> # process for admin
+        >>> UsgsNdviSmoothed.processed_file_suffix = "adm1"
+        >>> UsgsNdviSmoothed.process()
+        """
+        processed_path = self._get_processed_path()
+
+        # get dates for processing
+        process_dates = _expand_dekads(
+            y1=self._start_year,
+            d1=self._start_dekad,
+            y2=self._end_year,
+            d2=self._end_dekad,
+        )
+
+        # check to see if file exists and remove
+        # if clobber or check dates already processed
+        # if not
+        if processed_path.is_file():
+            df = self.load()
+
+            # check that the processed file has the same analyzed
+            # indicators and column for aggregation as passed
+            # to process()
+            cols = kwargs.get(
+                "stats_list", ["mean", "std", "min", "max", "sum", "count"]
+            )
+            percentile_list = kwargs.get("percentile_list")
+            if percentile_list is not None:
+                for percent in percentile_list:
+                    cols.append(f"{percent}quant")
```

Comment:
I looked into this, but couldn't figure out a convenient way to do a list comprehension on an existing list (since we are coming in with `cols` already filled out from above. Any suggestions?

Reply:
I think you should just be able to do `existing_list += [comprehension]`

Reply:
Oh wow. So simple and clean, of course you can. Great idea!!